tests/js/spec/components/smartSearchBar.spec.jsx

```diff
@@ -260,6 +277,28 @@ describe('SmartSearchBar', function() {
       await tick();
       expect(props.onSearch).toHaveBeenCalledWith('');
     });
+
+    it('invokes onSearch() on submit in hasPinnedSearch mode', function() {
+      const stubbedOnSearch = jest.fn();
+      const wrapper = mount(
+        <SmartSearchBar
+          onSearch={stubbedOnSearch}
+          organization={organization}
+          orgId="123"
+          projectId="456"
+          query="is:unresolved"
+          supportedTags={supportedTags}
+          hasPinnedSearch
+        />,
+        options
+      );
+
+      wrapper.find('form').simulate('submit', {
```

Comment:
Would simulating an enter keypress fire a form submit? Can we test for that instead?

Reply:
Sure thing. I was basing this off the existing tests, but I'll change it around.